ddtrace/contrib/django/utils.py

```diff
@@ -294,3 +295,19 @@ def _after_request_tags(pin, span, request, response):
     finally:
         if span.resource == REQUEST_DEFAULT_RESOURCE:
             span.resource = request.method
+
+
+class DjangoViewProxy(FunctionWrapper):
+    """
+    This custom function wrapper is used to wrap the callback passed to django views handlers (path/re_path/url).
+    This allows us to distinguish between wrapped django views and wrapped asgi applications in django channels.
+    """
+
+    @property
+    def __module__(self):
+        """
+        Overriding the __module__ property here is required to access the __module__ of a wrapped function when
+        __wrapped__.__module__ == None (this is the case for functools.partial()). In this case the __module__
+        of the wrapper is returned (ddtrace.contrib.django.utils) which effects the resource name of spans.
```

Comment:
I am having trouble understanding what you mean by this:

> required to access the __module__ of a wrapped function when __wrapped__.__module__ == None (this is the case for functools.partial())

Reply:
I reworded this section. Basically when the module attribute is not available the module name of the DjangoViewProxy will be resolved. This messes with resource names set for path/re_path/url handlers